playground/src/TabPane.tsx

```diff
@@ -18,11 +18,13 @@ export interface ITabPaneState {
 }
 
 export class TabPane extends React.Component<ITabPaneProps, ITabPaneState>  {
+  private DEFAULT_TAB_INDEX: number = 0;
+
   constructor(props: ITabPaneProps, context?: any) { // tslint:disable-line:no-any
     super(props, context);
 
     this.state = {
-      selectedTabIndex: 0
+      selectedTabIndex: document.location.hash ? this.getTabIndexFromUrl() : this.DEFAULT_TAB_INDEX
```

Comment:
Could be opt-in if you wish. Will see if feature is agreeable first.

Reply:
This feature is interesting, but what would be really awesome would be something like the "Share" button supported by TypeScript's playground.  Try this URL for example:  `[http://www.typescriptlang.org/play/#src=console.log(%22Hello%2C%20world!%22)`](http://www.typescriptlang.org/play/#src=console.log(%22Hello%2C%20world!%22)`)

Perhaps you could use a key so we can do e.g. `/tdoc/#tab=1&src=/**%20blah*/`

---
In reply to: [223574895](https://github.com/Microsoft/tsdoc/pull/95#discussion_r223574895) [](ancestors = 223574895)

Reply:
> 
> 
> This feature is interesting, but what would be really awesome would be something like the "Share" button supported by TypeScript's playground. Try this URL for example: `[http://www.typescriptlang.org/play/#src=console.log(%22Hello%2C%20world!%22)`]([http://www.typescriptlang.org/play/#src=console.log(%22Hello%2C%20world!%22)`](http://www.typescriptlang.org/play/#src=console.log(%22Hello%2C%20world!%22)%60))
> 
> Perhaps you could use a key so we can do e.g. `/tdoc/#tab=1&src=/**%20blah*/`
> 
> In reply to: [223574895](https://github.com/Microsoft/tsdoc/pull/95#discussion_r223574895) [](ancestors = 223574895)

:+1: to expanding this / future proofing to support serializing the input in the URL via query param

Re: index vs. title my concern would be bookmarks / share links breaking if the tab order is modified. Let's chat tomorrow.

Reply:
Import from file Github repo / Gist may also be useful